internal/controllers/reconciliation/controller.go

```diff
@@ -271,6 +271,30 @@ func (c *Controller) reconcileSnapshot(ctx context.Context, comp *apiv1.Composit
 
 	// Dry-run the update to see if it's needed
 	if !c.disableSSA {
+		// Before the dry-run, normalize conflicting field managers to "eno" to prevent SSA validation errors
+		// caused by multiple managers owning overlapping fields. When managers are renamed to "eno", the
+		// subsequent SSA Apply will treat eno as the sole owner and automatically merge the managedFields
+		// entries into a single consolidated entry for eno.
+		if current != nil {
+			wasModified, updatedManagers, err := resource.NormalizeConflictingManagers(current)
+			if err != nil {
+				return false, fmt.Errorf("normalize conflicting manager failed: %w", err)
+			}
+			if wasModified {
+				logger.Info("Normalized Conflicting Manager for eno", "updatedManager", updatedManagers, "managedFieldsBefore", current.GetManagedFields())
+				err = c.upstreamClient.Update(ctx, current)
```

Comment:
Update will update all the fields. If some fields are not specified explicitly, I would expect they'll be updated to default value. Could it be a side effect we don't expect?

Reply:
we do an update on `current` and `current` is the current representation of the k8s resource like `Deployment`, `DaemonSet`... so it will not modify the Spec or any other fields except for the managed Field entries. We normalize this field entry before the eno dry-run to avoid issues where we having legacy managers owning conflicting fields.